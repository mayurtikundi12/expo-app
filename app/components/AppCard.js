import { View, Image, StyleSheet, TouchableHighlight, Button } from 'react-native';
// import Swipeable from 'react-native-gesture-handler/Swipeable';

import React from 'react';

import AppColours from '../config/AppColours';
import AppText from './AppText';

function AppCard({title, subtitle, image, onPress}) {
    return (
        // <Swipeable renderRightActions={onSwipeLeft}>    
            <TouchableHighlight onPress={onPress} underlayColor={AppColours.white}>   
                <View style={styles.container}>
                    <Image source={image} style={styles.image}/>
                    <AppText style={styles.title}> {title} </AppText>
                    <AppText style={styles.subtitle}> {subtitle} </AppText>
                </View>
            </TouchableHighlight> 
        // </Swipeable>
    );
}
const styles = StyleSheet.create({
    container:{
        backgroundColor: AppColours.lightG,
        borderRadius: 20,
        overflow: "hidden",
        marginBottom: 25,
    },
    image: {
        height: 200,
        width: "100%",
    },
    title: {
        fontSize: 20,
        color: AppColours.white,
        fontFamily: "Roboto",
        alignItems: 'center',
        fontWeight: 'bold',
    },
    subtitle: {
        fontSize: 15,
        color: AppColours.white,
        fontFamily: "Roboto",
        alignItems: 'center',
    },
})

export default AppCard;