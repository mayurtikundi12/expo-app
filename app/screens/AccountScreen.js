import React from 'react';
import { View, StyleSheet } from 'react-native';
import {MaterialCommunityIcons} from '@expo/vector-icons';

import AppColours from '../config/AppColours';
import AppListItem from '../components/AppListItem';

function AccountScreen(props) {
    return (
        <View style={styles.container}>
            <View style={styles.welcomeContainer}>
                <MaterialCommunityIcons
                    name="airplane-takeoff"
                    size={60}
                    color={AppColours.darkG}/>
            </View>
            <View style={styles.profileContainer}>
             

                <AppListItem 
                
                title="BillieEilish" 
                subtitle="be@gmail.com"/>

            </View>
            
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        padding:20,
        backgroundColor:AppColours.lightG,
    },
    welcomeContainer:{
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 50,
    }, 
    profileContainer:{
        marginVertical:50,

    }
})

export default AccountScreen;