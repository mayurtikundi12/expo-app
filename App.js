import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import AuthNavigator from './app/navigation/AuthNavigator';
import { View,StyleSheet } from 'react-native';
import TravelScreen from './app/screens/TravelScreen';
import WelcomeScreen from './app/screens/WelcomeScreen';
import LoginScreen from './app/screens/LoginScreen';
import RegisterScreen from './app/screens/RegisterScreen';
import AppSwitch from './app/components/AppSwitch';
import AccountScreen from './app/screens/AccountScreen';
import { createStackNavigator } from '@react-navigation/stack';
import AppCard from './app/components/AppCard';

export default function App() {
  const AppStack = createStackNavigator();
  return (   

    <AccountScreen/>

    // <NavigationContainer>
    // <AppStack.Navigator initialRouteName="Welcome">
    //     <AppStack.Screen name="Welcome" component={WelcomeScreen} options={{headerShown:false}}/>
    //     <AppStack.Screen name="Login" component={LoginScreen} options={{title: 'Login Screen'}}/>
    //     <AppStack.Screen name="Register" component={RegisterScreen} options={{title: 'Sign Up for TravApp!'}}/>
    //     <AppStack.Screen name="travel" component={TravelScreen} />
    //     <AppStack.Screen name="Account" component={AccountScreen}/>

    // </AppStack.Navigator>
    // </NavigationContainer>
  );
}

