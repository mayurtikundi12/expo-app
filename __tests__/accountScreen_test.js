import React from 'react';
import App from '../App.js';
import 'react-native';
import renderer from 'react-test-renderer';
import {
    render,
    fireEvent,
    act
} from '@testing-library/react-native';
import AppText from '../app/components/AppText';
import AccountScreen from '../app/screens/AccountScreen.js';
import User from '../app/context/userContext';
import AppCard from '../app/components/AppCard.js';
describe('accountScreen test', () => {
    let tree, treeInstance;
    let userInfo = {"email":"a@a.com","username":"a"}
    beforeAll(async () => {

        tree = await renderer.create( 
            <User.Provider value={{userInfo}} >
            <AccountScreen/> 
            </User.Provider>
        )
        treeInstance = tree.root
    })

    it('Account Screen Test', async () => {
        expect(tree.toJSON()).toMatchSnapshot()
    })

    it('Delete Test', async() => {
        let button = treeInstance.findByProps({testID: 'TID9'})
        expect(button).toBeTruthy();
        await act(async()=>{
            await button.props.onPress();
        })
        let updatedButtons = treeInstance.findAllByProps({testID: 'TID9'})
        expect(updatedButtons.length).toEqual(0);
    });

    it('Adding New Place', async() => {
        let imageText = treeInstance.findByProps({testID: 'imageText'})
        let title = treeInstance.findByProps({testID: 'accountTitleText'})
        let subTitle = treeInstance.findByProps({testID: 'accountSubTitleText'})
        let btn =  treeInstance.findByProps({testID: 'AddPlace'});
        let originalCards = treeInstance.findAllByType(AppCard);
        await act(async ()=>{
            await fireEvent.changeText(imageText,'../../assets/bora-bora.jpg');
            await fireEvent.changeText(title,'New Place');
            await fireEvent.changeText(subTitle,'new Palce Sub title');
            await fireEvent(btn,'press');
        });

        let newCards = treeInstance.findAllByType(AppCard);
        expect(newCards.length).toEqual(originalCards.length+1);
    })

})


