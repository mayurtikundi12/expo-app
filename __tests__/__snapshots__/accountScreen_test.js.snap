// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accountScreen test Account Screen Test 1`] = `
<RCTScrollView
  pagingEnabled={true}
>
  <View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#1d7874",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginTop": 50,
            }
          }
        >
          <Text />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "fontWeight": "bold",
              "justifyContent": "space-between",
              "padding": 30,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 17,
                "height": 75,
                "width": 75,
                "zIndex": 10,
              }
            }
          />
          <View>
            <Image />
            <Text />
            <Text>
              a@a.com
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#1d7874",
              "borderRadius": 25,
              "flexDirection": "row",
              "marginVertical": 20,
              "padding": 10,
              "width": "100%",
            }
          }
        >
          <Text />
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Optima",
                  "fontSize": 20,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
             
             
            Categories
             
             
          </Text>
          <Text />
        </View>
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          visible={false}
        >
          <View>
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Close
                </Text>
              </View>
            </View>
            <RCTScrollView
              data={
                Array [
                  Object {
                    "backgroundColor": "red",
                    "icon": "flash",
                    "label": "Tourist spots",
                    "value": 1,
                  },
                  Object {
                    "backgroundColor": "red",
                    "icon": "flash",
                    "label": "Resorts",
                    "value": 2,
                  },
                  Object {
                    "backgroundColor": "red",
                    "icon": "flash",
                    "label": "Shop",
                    "value": 3,
                  },
                  Object {
                    "backgroundColor": "red",
                    "icon": "flash",
                    "label": "Things to do",
                    "value": 4,
                  },
                ]
              }
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": undefined,
                          "borderRadius": 20,
                          "height": 40,
                          "justifyContent": "center",
                          "width": 40,
                        }
                      }
                    >
                      <Text />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Optima",
                            "fontSize": 20,
                          },
                          undefined,
                        ]
                      }
                    >
                       
                      Tourist spots
                       
                    </Text>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": undefined,
                          "borderRadius": 20,
                          "height": 40,
                          "justifyContent": "center",
                          "width": 40,
                        }
                      }
                    >
                      <Text />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Optima",
                            "fontSize": 20,
                          },
                          undefined,
                        ]
                      }
                    >
                       
                      Resorts
                       
                    </Text>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": undefined,
                          "borderRadius": 20,
                          "height": 40,
                          "justifyContent": "center",
                          "width": 40,
                        }
                      }
                    >
                      <Text />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Optima",
                            "fontSize": 20,
                          },
                          undefined,
                        ]
                      }
                    >
                       
                      Shop
                       
                    </Text>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    focusable={false}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": undefined,
                          "borderRadius": 20,
                          "height": 40,
                          "justifyContent": "center",
                          "width": 40,
                        }
                      }
                    >
                      <Text />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "Optima",
                            "fontSize": 20,
                          },
                          undefined,
                        ]
                      }
                    >
                       
                      Things to do
                       
                    </Text>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </Modal>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Optima",
                "fontSize": 20,
              },
              Object {
                "alignItems": "center",
                "fontFamily": "Roboto",
                "fontSize": 25,
                "fontWeight": "bold",
              },
            ]
          }
        >
           
          Places you've visited in the past 
           
        </Text>
        <View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#1d7874",
                  "borderRadius": 20,
                  "marginBottom": 25,
                  "overflow": "hidden",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 200,
                    "width": "100%",
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Optima",
                      "fontSize": 20,
                    },
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                 
                 
                Bora Bora
                 
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Optima",
                      "fontSize": 20,
                    },
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "fontFamily": "Roboto",
                      "fontSize": 15,
                    },
                  ]
                }
              >
                 
                 
                Visited on 4th June 2019
                 
                 
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="TID8"
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                delete
              </Text>
            </View>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#1d7874",
                  "borderRadius": 20,
                  "marginBottom": 25,
                  "overflow": "hidden",
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 200,
                    "width": "100%",
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Optima",
                      "fontSize": 20,
                    },
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                 
                 
                Switzerland
                 
                 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Optima",
                      "fontSize": 20,
                    },
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "fontFamily": "Roboto",
                      "fontSize": 15,
                    },
                  ]
                }
              >
                 
                 
                Visted on 8th March 2018
                 
                 
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="TID9"
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                delete
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Image URl"
          rejectResponderTermination={true}
          testID="imageText"
          textContentType="imageText"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="Title"
          onChangeText={[Function]}
          placeholder="Title"
          rejectResponderTermination={true}
          testID="accountTitleText"
          textContentType="accountTitleText"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Sub Title"
          rejectResponderTermination={true}
          testID="accountSubTitleText"
          textContentType="accountSubTitleText"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="AddPlace"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add Place
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
